docs/DeveloperGuide.md

```diff
@@ -1,38 +1,177 @@
+6. Otherwise, the feedback will be an error message
+7. This method will return an object called `CommandResult` and pass `feedbackToUser` as its argument.
+8. `Duke` will subsequently call `printResult` method from `CommandResult`.
 
-{Describe the value proposition: what problem does it solve?}
+Class Diagram (to be added)
+Object Diagram (to be added)
 
-## User Stories
+### Find feature
+The find command is implemented using a `FindCommand` class which
+utilizes `findFood` method of the FoodList.
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+The `findFood` method implementation is as follows:
+1. Initialize `FoodList` result variable
+2. Start a loop iterating over each foodItem 
+3. Do word processing on a term and check if the foodItem name includes it
+4. Enter a loop with switch statement for each component of the `flags` argument 
+5. If some flag is not satisfied, continue the loop without adding foodItem to the result
+6. If the flags loop is over, the flags conditions were satisfied, so add the foodItem to the result
 
-## Non-Functional Requirements
+**Class Diagram** 
+![ClassDiagram](images/FindCommandClassDiagram.png)
 
-{Give non-functional requirements}
 
-## Glossary
+**Sequence Diagram**
+![ClassDiagram](images/FindCommandSequenceDiagram.png)
+
+### List feature
+The list command is implemented using a `ListCommand` class which utilizes `COMMAND_WORD`
+attributes of all other commands within the `commands` package.
 
-* *glossary item* - Definition
+The list command is implemented as follows:
+1. New `ListCommand` object created will call the method `execute` of `Command` class.
+2. The method `execute` decides what to append to the string `printToUser` according to `foodList`. 
+3. `foodList` is an object of type `FoodList` that is passed as argument to `execute`, it consists of the list of food 
+that has been added by user.
+4. `foodList` contains of `Food` that has undergoes the `toString` method.
+5. `execute` will check if `numberOfFood` is greater than 0, in which `numberOfFood` is obtained from the `FoodList` 
+function `getNumberOfFood`.
+6. If `numberOfFood` is greater than 0,  the foodList in String will be appended to `printToUser`. If it is not, then 
+the empty `foodList` will not be appended to `printToUser`.
+7. After appending the `foodList`, the method `execute` will return an object called `CommandResult` and pass          
+`printToUser` as its argument.
+8. `Duke` will then call `printResult` method from `CommandResult` which will print the food list for the user.
+
+**Class Diagram**
+![ClassDiagram](images/ListCommandClassDiagram.png)
+
+**Object Diagram**
```

Comment:
Is this a typo/inaccuracy? Diagram is a sequence diagram.
![image](https://user-images.githubusercontent.com/88440925/228427654-1ad988cd-44d2-465a-adc4-bb4b78ae11fb.png)